src/components/chips/chips.spec.js

```diff
@@ -213,20 +213,40 @@ describe('<md-chips>', function() {
         expect(scope.removeChip.calls.mostRecent().args[1]).toBe(0);       // Index
       });
 
+      var CHIP_COMPONENT_TEMPLATE =
+        '<md-chips ng-model="items" md-on-chip-select="onChipSelect($index)" md-on-chip-deselect="onChipDeselect($index)"></md-chips>';
 
-      it('should call the select method when selecting a chip', function() {
-        var element = buildChips(CHIP_SELECT_TEMPLATE);
+      it("should call the select and deselect expressions", function(done) {
+        var element = buildChips(CHIP_COMPONENT_TEMPLATE);
         var ctrl = element.controller('mdChips');
+        var scope = element.scope();
+        var state = 0;
```

Comment:
No, its not another test, it tests a series of events that should we triggered.

The test goes as following:

**Default Select `-1`** (normal behavior, called at compiling element)
- Calls Select Event (except new index to be `-1`) 

**Select `0`** (called at beginning of test)
- Calls Deselect Event (except old index to be `-1`)
- Calls Select Event (except new index to be `0`)

**Select `1`** (called after selecting `0`)
- Calls Deselect Event (except old index to be `0`)
- Calls Select Event (except new index to be `1`)

So this thing should definitely provided in one test (I hope you understand my thinks) :smile: